src/main/java/edu/cornell/kfs/module/ar/service/impl/CuCustomerAddressHelperServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package edu.cornell.kfs.module.ar.service.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.integration.cg.ContractsAndGrantsBillingAward;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.ar.ArConstants.ContractsAndGrantsInvoiceDocumentCreationProcessType;
+
+import edu.cornell.kfs.module.ar.service.CuCustomerAddressHelperService;
+
+public class CuCustomerAddressHelperServiceImpl implements CuCustomerAddressHelperService {
+    
+    private static final Logger LOG = LogManager.getLogger();
+    
+    /*
+     * CU Customization: KFSPTS-26393
+     * When creationProcessType is BATCH, validate that the customer number on the award as part if the customer
+     * address composite key matches the customer number associated to the agency associated to the award.
+     */
+    public boolean agencyCustomerMatchesAwardCustomer(ContractsAndGrantsBillingAward award,
+            ContractsAndGrantsInvoiceDocumentCreationProcessType creationProcessType) {
+        
+        if (ContractsAndGrantsInvoiceDocumentCreationProcessType.BATCH == creationProcessType) {
+            if (ObjectUtils.isNotNull(award.getAgency().getCustomer().getCustomerNumber()) &&
+                    ObjectUtils.isNotNull(award.getCustomerNumber()) &&
```

Comment:
I am not sure this check is needed if StringUtils.isNotBlank is already checking for null